io/io/inc/TThreadMergingFile.h

```diff
@@ -0,0 +1,41 @@
+#ifndef ROOT_TThreadMergingFile
+#define ROOT_TThreadMergingFile
+
+//////////////////////////////////////////////////////////////////////////
+//                                                                      //
+// TThreadMergingFile                                                   //
+//                                                                      //
+// Specialization of TMemFile to connect to a parallel file merger.     //
+// Upon a call to UploadAndReset, the content already written to the    //
+// file is upload to the server and the object implementing the function//
+// ResetAfterMerge (like TTree) are reset.                              //
+// The parallel file merger will then collate the information coming    //
+// from this client and any other client in to the file described by    //
+// the filename of this object.                                         //
+//                                                                      //
+//////////////////////////////////////////////////////////////////////////
+
+#include "dcqueue.h"
```

Comment:
It seems ROOT has no equivalent at this time, so we are thinking about using [`tbb::concurrent_queue`](https://software.intel.com/en-us/node/506200) instead of this one.